PlatformDynamicRangeLimit now has static initial values

squelart opened the pull request:
#### f4cf0e014e176898aed97ac0fc8d3f163775717d
<pre>
PlatformDynamicRangeLimit now has static initial values
<a href="https://bugs.webkit.org/show_bug.cgi?id=289685">https://bugs.webkit.org/show_bug.cgi?id=289685</a>
<a href="https://rdar.apple.com/146936785">rdar://146936785</a>

Reviewed by Mike Wyrzykowski and Anne van Kesteren.

PlatformDynamicRangeLimit doesn&apos;t have a default constructor anymore.
Users must explicitly pick an initial value, or construct another way
like Style::DynamicRangeLimit::toPlatformDynamicRangeLimit().

Note the distinct initial value for videos.

* Source/WebCore/html/HTMLMediaElement.h:
* Source/WebCore/platform/graphics/ImagePaintingOptions.h:
* Source/WebCore/platform/graphics/MediaPlayer.h:
* Source/WebCore/platform/graphics/PlatformDynamicRangeLimit.h:
(WebCore::PlatformDynamicRangeLimit::initialValue):
(WebCore::PlatformDynamicRangeLimit::initialValueForVideos):
* Tools/TestWebKitAPI/Tests/WebCore/PlatformDynamicRangeLimitTests.cpp:
(TestWebKitAPI::TEST(PlatformDynamicRangeLimit, Values)):
(TestWebKitAPI::TEST(PlatformDynamicRangeLimit, StaticValues)):
(TestWebKitAPI::TEST(PlatformDynamicRangeLimit, DefaultConstruction)): Deleted.

Canonical link: <a href="https://commits.webkit.org/292117@main">https://commits.webkit.org/292117@main</a>
</pre>
<!--EWS-Status-Bubble-Start-->
https://github.com/WebKit/WebKit/commit/b9f431d7c9d7042ea934b671bb6397e094cfa8c9

| Misc | iOS, visionOS, tvOS & watchOS  | macOS  | Linux |  Windows |
| ----- | ---------------------- | ------- |  ----- |  --------- |
| [✅ 🧪 style](https://ews-build.webkit.org/#/builders/38/builds/94905 "Passed style check") | [✅ 🛠 ios](https://ews-build.webkit.org/#/builders/131/builds/14496 "Built successfully") | [✅ 🛠 mac](https://ews-build.webkit.org/#/builders/138/builds/4342 "Built successfully") | [✅ 🛠 wpe](https://ews-build.webkit.org/#/builders/5/builds/99924 "Built successfully") | [✅ 🛠 win](https://ews-build.webkit.org/#/builders/59/builds/45402 "Built successfully") 
| [✅ 🧪 bindings](https://ews-build.webkit.org/#/builders/9/builds/96952 "Passed tests") | [✅ 🛠 ios-sim](https://ews-build.webkit.org/#/builders/130/builds/14779 "Built successfully") | [✅ 🛠 mac-AS-debug](https://ews-build.webkit.org/#/builders/123/builds/22916 "Built successfully") | [✅ 🧪 wpe-wk2](https://ews-build.webkit.org/#/builders/34/builds/72397 "Passed tests") | [✅ 🧪 win-tests](https://ews-build.webkit.org/#/builders/60/builds/29691 "Passed tests") 
| [✅ 🧪 webkitperl](https://ews-build.webkit.org/#/builders/11/builds/97906 "Passed tests") | [✅ 🧪 ios-wk2](https://ews-build.webkit.org/#/builders/132/builds/11014 "Passed tests") | [✅ 🧪 api-mac](https://ews-build.webkit.org/#/builders/18/builds/85692 "Passed tests") | [❌ 🧪 api-wpe](https://ews-build.webkit.org/#/builders/41/builds/52728 "Found 1 new API test failure: /TestWebCore:GStreamerTest.capsFromCodecString (failure)") | 
| | [✅ 🧪 ios-wk2-wpt](https://ews-build.webkit.org/#/builders/133/builds/10721 "Passed tests") | [✅ 🧪 mac-wk1](https://ews-build.webkit.org/#/builders/135/builds/3411 "Passed tests") | [✅ 🛠 wpe-cairo](https://ews-build.webkit.org/#/builders/65/builds/44735 "Built successfully") | 
| | [  ~~🧪 api-ios~~](https://ews-build.webkit.org/#/builders/13/builds/80966 "The change is no longer eligible for processing. Commit was outdated when EWS attempted to process it.") | [✅ 🧪 mac-wk2](https://ews-build.webkit.org/#/builders/136/builds/3505 "Passed tests") | [✅ 🛠 gtk](https://ews-build.webkit.org/#/builders/2/builds/101965 "Built successfully") | 
| | [✅ 🛠 vision](https://ews-build.webkit.org/#/builders/128/builds/21937 "Built successfully") | [✅ 🧪 mac-AS-debug-wk2](https://ews-build.webkit.org/#/builders/122/builds/16018 "Passed tests") | [✅ 🧪 gtk-wk2](https://ews-build.webkit.org/#/builders/1/builds/81392 "Passed tests") | 
| | [✅ 🛠 vision-sim](https://ews-build.webkit.org/#/builders/121/builds/22184 "Built successfully") | [✅ 🧪 mac-wk2-stress](https://ews-build.webkit.org/#/builders/8/builds/81716 "Passed tests") | [❌ 🧪 api-gtk](https://ews-build.webkit.org/#/builders/21/builds/80783 "Found 1 new API test failure: /TestWebCore:GStreamerTest.capsFromCodecString (failure)") | 
| [✅ 🛠 🧪 merge](https://ews-build.webkit.org/#/builders/19/builds/20203 "Built successfully and passed tests") | [✅ 🧪 vision-wk2](https://ews-build.webkit.org/#/builders/126/builds/25342 "Passed tests") | [✅ 🧪 mac-intel-wk2](https://ews-build.webkit.org/#/builders/137/builds/2736 "Passed tests") | [✅ 🛠 playstation](https://ews-build.webkit.org/#/builders/134/builds/15175 "Built successfully") | 
| | [✅ 🛠 tv](https://ews-build.webkit.org/#/builders/127/builds/21913 "Built successfully") | [✅ 🛠 mac-safer-cpp](https://ews-build.webkit.org/#/builders/120/builds/27030 "Built successfully") | | 
| | [✅ 🛠 tv-sim](https://ews-build.webkit.org/#/builders/125/builds/21572 "Built successfully") | | | 
| | [✅ 🛠 watch](https://ews-build.webkit.org/#/builders/129/builds/25044 "Built successfully") | | | 
| | [✅ 🛠 watch-sim](https://ews-build.webkit.org/#/builders/124/builds/23313 "Built successfully") | | | 
<!--EWS-Status-Bubble-End-->

weinig reviewed (changes requested):
There is nothing in the current specs that would have different values for videos. What is guiding this change?

squelart committed:
PlatformDynamicRangeLimit now has static initial values
https://bugs.webkit.org/show_bug.cgi?id=289685
rdar://146936785

Reviewed by Mike Wyrzykowski and Anne van Kesteren.

PlatformDynamicRangeLimit doesn't have a default constructor anymore.
Users must explicitly pick an initial value, or construct another way
like Style::DynamicRangeLimit::toPlatformDynamicRangeLimit().

Note the distinct initial value for videos.

* Source/WebCore/html/HTMLMediaElement.h:
* Source/WebCore/platform/graphics/ImagePaintingOptions.h:
* Source/WebCore/platform/graphics/MediaPlayer.h:
* Source/WebCore/platform/graphics/PlatformDynamicRangeLimit.h:
(WebCore::PlatformDynamicRangeLimit::initialValue):
(WebCore::PlatformDynamicRangeLimit::initialValueForVideos):
* Tools/TestWebKitAPI/Tests/WebCore/PlatformDynamicRangeLimitTests.cpp:
(TestWebKitAPI::TEST(PlatformDynamicRangeLimit, Values)):
(TestWebKitAPI::TEST(PlatformDynamicRangeLimit, StaticValues)):
(TestWebKitAPI::TEST(PlatformDynamicRangeLimit, DefaultConstruction)): Deleted.

Canonical link: https://commits.webkit.org/292117@main
